### PR TITLE
feat: Settings Menu

### LINKED ISSUE #530: Bind Landscape/LODs to Quality Settings
### Problem
The quality settings are only bound to the graphics, but none of the parametrization of the environment (load radius, environment details, LOD thresholds) are connected to them, and need to be changed manually on every run.

### Solution
Associate environments, scene load radius, and LOD thresholds to Quality settings. 
Have a simple way to modify those values on editor to tweak them based on different devices performance.

High
- Scene Load Radius: 150
- LOD1 Threshold: 2
- LOD2 Threshold: 4
- Terrain LOD Bias: 250
- Detail Density: 100
- Grass Distance: 300
- Chunk Cull Distance: 7000

Medium
- Scene Load Radius: 50
- LOD1 Threshold: 1
- LOD2 Threshold: 3
- Terran LOD Bias: 150
- Detail Density: 70
- Grass Distance: 150
- Chunk Cull Distance: 3000

Low
- Scene Load Radius: 20
- LOD1 Threshold: 1
- LOD2 Threshold: 1
- Terrain LOD Bias: 50
- Detail Density: 30
- Grass Distance: 75
- Chunk Cull Distance: 1000

### PR DESCRIPTION
**Pull request by sandrade-dcl**
### What this PR includes?
- Implementation of a dynamic Settings Menu.
- **GRAPHICS** section:
  - Addition of **Graphics Quality** dropdown setting (Fix #530 ).
  - Addition of **Scene Distance** slider setting.
  - Addition of **Environment Distance** slider setting.

image

### What this PR DOESN'T includes?
- **SOUND** section: The corresponding controls are placed in the menu but without implementation (it will be implemented in a separated PR).
- **CONTROLS** section: The corresponding controls are placed in the menu but without implementation (it will be implemented in a separated PR).
- Etc...

### CODE REVIEW

**Review by mikhail-dcl — DISMISSED** [review]
Everything looks great just like we discussed in the beginning 💪 
I have only minor requests

File: Explorer/Assets/DCL/Settings/Configuration/SettingsMenuConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL.Settings.Configuration
+{
+    [CreateAssetMenu(menuName = "Settings Menu/Create Settings Menu Configuration", fileName = "Settings Menu Configuration", order = 0)]
+    public class SettingsMenuConfiguration : ScriptableObject
+    {
+        [field: SerializeField] public SettingsGroupView SettingsGroupPrefab { get; private set; }
+        [field: SerializeField] public SettingsSectionConfig GeneralSectionConfig { get; set; }
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply Explorer/Assets/DCL/Settings/Configuration/SettingsMenuConfiguration.cs]: Done: https://github.com/decentraland/unity-explorer/pull/492/commits/8e733335a072d72231d286edff07574b8f6f4d42

File: Explorer/Assets/DCL/Settings/Configuration/SettingsModuleBindings.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                    return new ChatSoundsSettingsController(viewInstance);
+                // add other cases...
+            }
+
+            throw new ArgumentOutOfRangeException(nameof(viewInstance));
+        }
+    }
+
+    [Serializable]
+    public class SliderModuleBinding : SettingsModuleBinding<SettingsSliderModuleView, SettingsSliderModuleView.Config, SliderModuleBinding.SliderFeatures>
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply Explorer/Assets/DCL/Settings/Configuration/SettingsModuleBindings.cs]: Done: https://github.com/decentraland/unity-explorer/pull/492/commits/da73ce41be5562325361d1f30633baaaa2f3500b

File: Explorer/Assets/DCL/Settings/SettingsView.cs [hunk 3]
```diff
@@ -1,9 +1,33 @@
...
+using DCL.UI;
 using UnityEngine;
+using UnityEngine.UI;
 
 namespace DCL.Settings
 {
     public class SettingsView : MonoBehaviour
     {
+        [field: Header("Configuration")]
+        [field: SerializeField] public SettingsMenuConfiguration Configuration { get; private set; }
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply Explorer/Assets/DCL/Settings/SettingsView.cs]: Done: https://github.com/decentraland/unity-explorer/pull/492/commits/f8867aad51cce8de7e1c9ef6e53271ca83720e2f

**Review by m3taphysics — APPROVED** [review]

**Pull request merged**